calculator-app/src/Calculator.js

```diff
@@ -0,0 +1,219 @@
+      } else if (operation !== "") {
+        this.setState((prevState) => ({
+          display: "" + userInput,
+          history: prevState.history + userInput,
+        }));
+      }
+  };
+
+  handleOperand = (userInput) => {
+    const { operation, display } = this.state;
+    if (operation !== "") {
+      this.setState({ operation: userInput });
+    } else {
+      this.setState((prevState) => ({
+        operation: userInput,
+        history: prevState.display + userInput,
+        previousDisplay: display,
+      }));
+    }
+  };
+
+  handleEqual = () => {
+    const { previousDisplay, display, operation } = this.state;
+    // add check for commas(,)
+    this.setState((prevState) => ({
+      previousDisplay: "",
+      history: prevState.history + "=",
+      operation: "=",
+    }));
+    switch (operation) {
+      case "+":
+        this.setState({
+          display: Number(previousDisplay) + Number(display),
+        });
+        break;
+      case "-":
+        this.setState({
+          display: Number(previousDisplay) - Number(display),
+        });
+        break;
+      case "x":
+        this.setState({
+          display: Number(previousDisplay) * Number(display),
+        });
+        break;
+      case "÷":
+        this.setState({
+          display: Number(previousDisplay) / Number(display),
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  handleDecimal = () => {
+    const { previousDisplay } = this.state;
+    if (!previousDisplay.includes(".")) {
+      this.setState((prevState) => ({
+        previousDisplay: prevState.previousDisplay + ".",
+        display: prevState.display + ".",
+        history: prevState.history + ".",
+      }));
+    }
+  };
+
+  handleDelete = () => {
+    const { previousDisplay, display, history } = this.state;
+    if (previousDisplay === "") {
+      this.setState({
+        previousDisplay: "",
+      });
+    } else {
+      this.setState({
+        previousDisplay: previousDisplay.slice(0, -1),
+        display: display.slice(0, -1),
+        history: history.slice(0, -1),
+      });
+    }
+  };
+
+  handleAllClear = () => {
+    this.setState({
+      previousDisplay: "",
+      display: "0",
+      history: "",
+      operation: "",
+      sign: "positive",
+    });
+  };
+
+  handleSign = () => {
+    const { sign, previousDisplay, history, display } = this.state;
+    if (previousDisplay !== "") {
+      if (sign === "positive") {
+        this.setState({
+          sign: "negative",
+          previousDisplay: previousDisplay * -1,
+          display: display * -1,
+          history: history * -1,
+        });
+      } else {
+        this.setState({
+          sign: "positive",
+          previousDisplay: previousDisplay * -1,
+          display: display * -1,
+          history: history * -1,
+        });
+      }
+    }
+  };
+
+  handleZero = () => {
+    const { previousDisplay } = this.state;
+    if (previousDisplay !== "0") {
```

Comment:
This looks like it is not going to hit after I hit 2 (with the goal of getting 20)